### PR TITLE
[PLA-1648] add token collection name

### PR DESCRIPTION
**Pull request by zlayine**
### **PR Type**
Enhancement

___

### **Description**
- Added new columns for displaying collection and token names in the respective tables.
- Implemented methods to fetch and display names for collections and tokens.
- Added a new method `fetchUrl` in the `ApiService` to fetch data from a given URL.
- Renamed `fetchURL` to `fetchPlatformURL` and updated its usage in the store.

___

### **Changes walkthrough** 📝
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Enhancement
</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>Collections.vue</strong><dd><code>Add and display collection names in Collections table.</code>&nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

resources/js/components/pages/Collections.vue
<li>Added a new column for collection names in the table.<br> <li> Implemented logic to fetch and display collection names.<br> <li> Introduced methods <code>getCollectionName</code>, <code>setCollectionNames</code>, and <code>fetchUri</code> <br>to handle name retrieval.<br>

</details>
    

  </td>
  <td><a href="https://github.com/enjin/platform-ui/pull/121/files#diff-fa057c3b15bcc7a3a36a2664cdf4d7aecc0f7d2abfb6d89c0fa9a2d1e3f6ff2e">+51/-1</a>&nbsp; &nbsp; </td>
</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>Tokens.vue</strong><dd><code>Add and display token names in Tokens table.</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

resources/js/components/pages/Tokens.vue
<li>Added a new column for token names in the table.<br> <li> Implemented logic to fetch and display token names.<br> <li> Introduced methods <code>getTokenName</code>, <code>setTokenNames</code>, and <code>fetchUri</code> to handle <br>name retrieval.<br>

</details>
    

  </td>
  <td><a href="https://github.com/enjin/platform-ui/pull/121/files#diff-d68dcf9dc7162ad3ed194ff25f6b7f79b39a8a09cad90357dab2b7940e2fa966">+51/-1</a>&nbsp; &nbsp; </td>
</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>index.ts</strong><dd><code>Add fetchUrl method and rename fetchURL.</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

resources/js/api/index.ts
<li>Added a new method <code>fetchUrl</code> to fetch data from a given URL.<br> <li> Renamed <code>fetchURL</code> to <code>fetchPlatformURL</code>.<br>

</details>
    

  </td>
  <td><a href="https://github.com/enjin/platform-ui/pull/121/files#diff-eed1698e4da8153782b2abcbc87ffce6862ee99aa603127ae32392ed96ee8827">+10/-1</a>&nbsp; &nbsp; </td>
</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>index.ts</strong><dd><code>Update method call to fetchPlatformURL.</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

resources/js/store/index.ts
- Updated method call from `fetchURL` to `fetchPlatformURL`.

</details>
    

  </td>
  <td><a href="https://github.com/enjin/platform-ui/pull/121/files#diff-45cbdaed3eae4593b455daabf8cbc593b4d1f00058757cf6ca25a1b9967ebda2">+1/-1</a>&nbsp; &nbsp; &nbsp; </td>
</tr>                    
</table></td></tr></tr></tbody></table>

___

> 💡 **PR-Agent usage**:
>Comment `/help` on the PR to get a list of all available PR-Agent tools and their descriptions

### CODE REVIEW

File: resources/js/components/pages/Collections.vue [hunk 1]
```diff
@@ -283,6 +332,7 @@ const loadMoreCollectionsWithObserver = () => {
                     const res = await CollectionApi.getCollections(collections.value.cursor);
                     const data = DTOFactory.forCollections(res);
```
Comment (enjinabner) [inline resources/js/components/pages/Collections.vue:333]: Can't we reuse the data here to generate the names? so we don't need to do an extra api calls

**Review by enjinabner — CHANGES_REQUESTED** [review]
suggestion

File: resources/js/components/pages/Collections.vue (hunk 1 above)
Comment (enjinabner) [reply resources/js/components/pages/Collections.vue:333]: The attribute value was parsed in the BE already https://github.com/enjin/platform-core/blob/master/src/Models/Laravel/Attribute.php#L54

Comment (zlayine) [reply resources/js/components/pages/Collections.vue:333]: if there is URI, i dont think we can get it from somewhere else, does platform provide collection metadata ?

Comment (enjinabner) [reply resources/js/components/pages/Collections.vue:333]: Right, don't think its being parsed in the BE

**Review by enjinabner — APPROVED** [review]

**Pull request merged**